documentation/library_guide/parallel_api/range_based_api.rst

```diff
@@ -1,32 +1,75 @@
+
+* ``views::iota``: A range factory that generates a sequence of elements by repeatedly incrementing an initial value.
+* ``views::all``: A custom utility that represents a view of all or a part of ``sycl::buffer`` elements
+  for reading and writing on a device.
+* ``views::all_read``: A custom utility that represents a view of all or a part of ``sycl::buffer`` elements
+  for reading on a device.
+* ``views::all_write``: A custom utility that represents a view of all or a part of ``sycl::buffer`` elements
+  for writing on a device.
+* ``views::host_all``: A custom utility that represents a view of all or a part of ``sycl::buffer`` elements
+  for reading and writing on the host.
+* ``views::subrange``: A utility that represents a view of unified shared memory (USM) data range
+  defined by two USM pointers.
+* ``views::zip``: A custom range adaptor that produces one ``zip_view`` from other several views.
+* ``views::transform``: A range adaptor that represents a view of an underlying sequence after applying
+  a transformation to each element.
+* ``views::reverse``: A range adaptor that produces a reversed sequence of elements provided by another view.
+* ``views::take``: A range adaptor that produces a view of the first N elements from another view.
+* ``views::drop``: A range adaptor that produces a view excluding the first N elements from another view.
+
+Only these ranges, ``sycl::buffer``, and their combinations can be passed to the experimental range-based algorithms.
+
+A ``sycl::buffer`` wrapped with ``views::all`` and similar utilities, ``views::subrange`` over USM, and ``views::iota``
+are considered *base ranges*. The range adaptors may be combined into a pipeline with a base range at the beginning.
+For example:
 
 .. code:: cpp
 
     sycl::buffer<int> buf(data, sycl::range<1>(10));
-    auto range_1 = iota_view(0, 10) | views::reverse();
-    auto range_2 = all_view(buf) | views::reverse();
+    auto range_1 = views::iota(0, 10) | views::reverse;
+    auto range_2 = views::all(buf) | views::take(10);
+
+For ranges based on a SYCL buffer, data access is only permitted on a device, while ``size()`` and ``empty()``
+methods are allowed to be called on both host and device.
+
+Range-Based Algorithms
+----------------------
+
+The functions for experimental range based algorithms resemble the standard C++ parallel algorithm overloads
+where all data sequences represented by ranges instead of iterators or iterator pairs. For example,
 
-For the range, based on the ``all_view`` factory, data access is permitted on a device only. ``size()`` and ``empty()`` methods are allowed 
-to be called on both host and device.
+.. code:: cpp
+
+   template <typename ExecutionPolicy, typename Range1, typename Range2>
+   void copy(ExecutionPolicy&& exec, Range1&& source, Range2&& destination);
+
+where ``source`` is used instead of two iterators to represent the input, and ``destination`` represents the output.
```

Comment:
Please capitalize "where". I know that it continues the idea above, but that is solved with the colon instead of a comma and aligns with the style guide.

Reply:
These style rules are sometimes quite strange :)

Reply:
I agree!